### PR TITLE
Flow type components

### PR DESCRIPTION
**Pull request by landonreed**
Add flow typing to library (nearly) all components.  Notable components this avoids are some of the modal components, which need to be refactored altogether because they rely heavily on refs and accept state/props through their `open` method, and the Settings components, which should see a refactor in #41.

This PR raises the flow coverage to a semi-respectable 76%.

### CODE REVIEW

File: lib/alerts/components/AffectedServices.js [hunk 1]
```diff
@@ -86,36 +111,39 @@ export default class AffectedServices extends Component {
   }
 }
 
-class ServicesHeader extends Component {
+class ServicesHeader extends Component<{entities: Array<any>}> {
```
Comment (evansiroky) [inline lib/alerts/components/AffectedServices.js]: Should this be `Array<AlertEntity>` instead?  Seems like that would then negate the need for typing within each filter function.

File: lib/common/components/MapModal.js [hunk 2]
```diff
@@ -36,33 +63,33 @@ export default class MapModal extends Component {
...
+  _onEditPath () {
     console.log('Path edited !')
   }
 
-  _onCreate (e) {
-    polyline = e.layer
-    this.setState({rectangle: e.layer})
-    // To edit this polyline call : polyline.handler.enable()
-    console.log('Path created !', polyline)
+  // _onCreate (e) {
```
Comment (evansiroky) [inline lib/common/components/MapModal.js]: If we still need this even with comments, please indicate why and perhaps create a github issue.

File: lib/common/user/Auth0Manager.js [hunk 3]
```diff
@@ -194,6 +194,7 @@ class Auth0Manager {
     // Get profile with access token and return profile and ID token to store.
     return this._getProfileFromToken(accessToken)
       .then(profile => {
+        console.log(profile)
```
Comment (evansiroky) [inline lib/common/user/Auth0Manager.js]: remove

File: lib/editor/actions/active.js [hunk 4]
```diff
@@ -116,7 +116,7 @@ export function setActiveGtfsEntity (
     const {locationBeforeTransitions} = getState().routing
     const pathname = locationBeforeTransitions && locationBeforeTransitions.pathname
     if (!locationBeforeTransitions || !pathname || pathname !== url) {
-      // console.log('updating url', url, pathname)
+      console.log('updating url', url, pathname)
```
Comment (evansiroky) [inline lib/editor/actions/active.js]: remove

File: lib/editor/components/EntityListSecondaryActions.js [hunk 5]
```diff
@@ -28,7 +33,18 @@ export default class EntityListSecondaryActions extends Component {
...
+    focusedOptionIndex,
+    focusOption,
+    key,
+    labelKey,
+    option,
+    options,
+    selectValue,
+    style,
+    valueArray
+  }: any) => {
```
Comment (evansiroky) [inline lib/editor/components/EntityListSecondaryActions.js]: elaborate a bit on this type

File: lib/editor/components/FareRuleSelections.js [hunk 6]
```diff
@@ -1,44 +1,48 @@
...
+  activeComponent: string,
+  zones: any,
+  zoneOptions: Array<any>,
+  routeEntity: ?GtfsRoute,
+  updateActiveEntity: (Entity, string, any) => void
+}
 
-  _onContainsChange = (input) => {
+export default class FareRuleSelections extends Component<Props> {
+  _onContainsChange = (input: any) => {
```
Comment (evansiroky) [inline lib/editor/components/FareRuleSelections.js]: Please elaborate a bit on the types of these change functions

File: lib/editor/components/GtfsEditor.js [hunk 7]
```diff
@@ -18,19 +18,68 @@ import {getConfigProperty} from '../../common/util/config'
...
+  tripPatterns: Array<Pattern>,
+  fetchTripPatterns: () => void,
+  updateMapSetting: ({bounds?: any, target?: number, zoom?: number}) => void,
+  removeControlPoint: () => void,
+  updateUserMetadata: (any, any) => void,
+  updatePatternStops: (Pattern, Array<PatternStop>) => void,
+  handleControlPointDrag: () => void,
+  handleControlPointDragEnd: () => void,
+  handleControlPointDragStart: () => void,
+  setActivePatternSegment: () => void,
```
Comment (evansiroky) [inline lib/editor/components/GtfsEditor.js]: let's alphabetize these.  Try using [this plugin](https://atom.io/packages/sorter).

File: lib/editor/components/map/EditorMap.js [hunk 8]
```diff
@@ -6,19 +8,94 @@ import AddableStopsLayer from './AddableStopsLayer'
...
+  activeEntityId: number,
+  activeComponent: string,
+  subComponent: string,
+  currentPattern: Pattern,
+  stops: Array<GtfsStop>,
+  editSettings: EditSettingsState,
+  mapState: MapState,
+  feedSource: Feed,
+  feedInfo: FeedInfo,
+  zoomToTarget: ?number,
```
Comment (evansiroky) [inline lib/editor/components/map/EditorMap.js]: Please alphabetize this.

File: lib/editor/components/map/EditorMap.js [hunk 9]
```diff
@@ -95,40 +173,36 @@ export default class EditorMap extends Component {
...
-      const selectedStop = entities.find(stop => Math.round(stop.stop_lat * 1000) / 1000 === Math.round(e.latlng.lat * 1000) / 1000 && Math.round(stop.stop_lon * 1000) / 1000 === Math.round(e.latlng.lng * 1000) / 1000)
-      if (selectedStop) {
-        if (activeEntity && activeEntity.id === selectedStop.id) {
-          // do nothing, the user just clicked the current stop
-        } else {
-          setActiveEntity(feedSource.id, activeComponent, selectedStop)
-        }
-      }
-    }
+    // if (activeComponent === 'stop') {
```
Comment (evansiroky) [inline lib/editor/components/map/EditorMap.js]: Please add a comment explaining why this code is being kept.

File: lib/editor/components/map/EditorMap.js [hunk 10]
```diff
@@ -167,114 +241,111 @@ export default class EditorMap extends Component {
...
+              mapState={this.props.mapState}
               patternSegment={this.props.patternSegment}
-              editSettings={editSettings}
+              editSettings={this.props.editSettings}
               patternCoordinates={patternCoordinates} />
-            {editSettings.editGeometry &&
+            {this.props.editSettings.editGeometry &&
               <ControlPointsLayer
-                activePattern={activePattern}
+                activePattern={this.props.activePattern}
```
Comment (evansiroky) [inline lib/editor/components/map/EditorMap.js]: Please destructure out all these props.

File: lib/editor/components/pattern/TripPatternViewer.js [hunk 11]
```diff
@@ -23,7 +27,17 @@ export default class TripPatternViewer extends Component {
           maxLength={28}
           onChange={this.savePatternName} />
         <EditShapePanel {...this.props} />
-        <EditSchedulePanel {...this.props} />
+        <EditSchedulePanel
+          activeEntity={this.props.activeEntity}
+          activePattern={this.props.activePattern}
+          activePatternId={this.props.activePatternId}
+          activePatternTripCount={this.props.activePatternTripCount}
```
Comment (evansiroky) [inline lib/editor/components/pattern/TripPatternViewer.js]: Please destructure

File: lib/editor/components/timetable/EditableCell.js [hunk 12]
```diff
@@ -1,30 +1,64 @@
...
+  cellRenderer: (TimetableColumn, any) => number | string,
+  column: TimetableColumn,
+  columnIndex: number,
+  handlePastedRows: (Array<any>, number, number) => void,
+  invalidData: boolean,
+  isFocused: boolean,
+  isSelected: boolean,
+  lightText: boolean,
+  onChange: (any, number, TimetableColumn, number) => void,
+  renderTime: boolean,
```
Comment (evansiroky) [inline lib/editor/components/timetable/EditableCell.js]: Please alphabetize

File: lib/editor/components/timetable/Timetable.js [hunk 13]
```diff
@@ -7,23 +9,46 @@ import TimetableGrid from './TimetableGrid'
...
+  activeScheduleId: string,
+  addNewRow: (?boolean, ?boolean) => void,
+  cloneSelectedTrips: () => void,
+  offsetWithDefaults: (boolean) => void,
+  removeSelectedRows: () => void,
+  setActiveCell: ?string => void,
+  setOffset: number => void,
+  showHelpModal: () => void,
+  timetable: TimetableState,
+  columns: Array<TimetableColumn>,
```
Comment (evansiroky) [inline lib/editor/components/timetable/Timetable.js]: Alphabetize

File: lib/editor/components/timetable/TimetableEditor.js [hunk 14]
```diff
@@ -14,21 +14,30 @@ import Timetable from './Timetable'
...
   deleteTripsForCalendar: (string, Pattern, string, Array<Trip>) => void,
+  duplicateRows: () => void,
+  toggleDepartureTimes: () => void,
+  fetchCalendarTripCountsForPattern: () => void,
   feedSource: Feed,
   fetchTripsForCalendar: (string, Pattern, string) => void,
   offsetRows: ({rowIndexes: Array<any>, offset: number}) => void,
   saveTripsForCalendar: (string, Pattern, string, Array<Trip>) => void,
   setActiveEntity: (string, string, any) => void,
+  setActiveCell: ?string => void,
```
Comment (evansiroky) [inline lib/editor/components/timetable/TimetableEditor.js]: Alphabetize

File: lib/editor/components/timetable/TimetableGrid.js [hunk 15]
```diff
@@ -278,15 +303,15 @@ export default class TimetableGrid extends Component {
     }
   }
 
-  _renderLeftHeaderCell = ({ columnIndex, key, rowIndex, style }) => (
+  _renderLeftHeaderCell = ({ columnIndex, key, rowIndex, style }: { columnIndex: number, key: string, rowIndex: number, style: Style}) => (
```
Comment (evansiroky) [inline lib/editor/components/timetable/TimetableGrid.js]: Let's unwrap these to be more readable

File: lib/editor/components/timetable/TimetableHeader.js [hunk 16]
```diff
@@ -21,38 +23,46 @@ import RouteSelect from './RouteSelect'
...
+
+type Props = {
+  cloneSelectedTrips: () => void,
+  feedSource: Feed,
+  fetchCalendarTripCountsForPattern: () => void,
+  timetable: TimetableState,
+  setOffset: number => void,
+  offsetWithDefaults: (boolean) => void,
+  toggleDepartureTimes: () => void,
+  addNewRow: (?boolean, ?boolean) => void,
```
Comment (evansiroky) [inline lib/editor/components/timetable/TimetableHeader.js]: Alphabetize

File: lib/gtfsplus/components/GtfsPlusTable.js [hunk 17]
```diff
@@ -1,29 +1,35 @@
...
+type Props = {
+  newRowClicked: ({tableId: string}) => void,
+  currentPage: number,
+  deleteRowClicked: (string, number) => void,
+  feedSource: Feed,
+  fieldEdited: ({tableId: string, rowIndex: number, fieldName: string, newValue: any}) => void,
+  gtfsEntitySelected: (string, Entity) => void,
+  getGtfsEntity: (string, string) => Entity,
+  newRowsDisplayed: Array<any> => void,
+  pageCount: number,
```
Comment (evansiroky) [inline lib/gtfsplus/components/GtfsPlusTable.js]: Alphabetize

File: lib/manager/components/FeedSourceTable.js [hunk 18]
```diff
@@ -92,20 +103,23 @@ export default class FeedSourceTable extends Component {
...
-    user: PropTypes.object,
+type RowProps = {
+  feedSource: Feed,
+  hoverComponent: any,
+  project: Project,
+  user: UserState,
+  updateFeedSource: (Feed, any) => void,
+  saveFeedSource: (any) => void,
+  onHover: (?Feed) => void,
+  active: boolean,
```
Comment (evansiroky) [inline lib/manager/components/FeedSourceTable.js]: Alphabetize

**Review by evansiroky — CHANGES_REQUESTED** [review]
Lots of great work here.  I think we should enable this eslint rule: https://github.com/gajus/eslint-plugin-flowtype#eslint-plugin-flowtype-rules-sort-keys  That way we can quickly alphabetize all these type definitions.

File: lib/editor/components/map/EditorMap.js (hunk 9 above)
Comment (landonreed) [reply lib/editor/components/map/EditorMap.js]: I will remove the code entirely.  The click to select stop functionality was essentially completely hidden from users and not entirely reliable or necessary.

**Comment by landonreed** [comment]
OK, @evansiroky.  I've addressed all of your comments.  Good to merge?

**Review by evansiroky — APPROVED** [review]
Great work and let's keep up the trend of keeping the e2e tests passing!

**Pull request merged**